Changelog updates (#684)
openownership/data-standard: docs/standard/changelog.rst

.. include:: warningbox.rst

0.4 - 2024-05-04
================ 

Added
-----
* Separation of metadata and data into statements and records
* ``declaration`` and ``declarationSubject`` to model declarations
* ``recordId`` and ``recordStatus`` to model change over time 
* New documentation pages: 

  * Dates guidance 
  * Generating statements
  * Record identifiers
  * Representing trusts 
  * Representing nominees 
  * Representing record updates

Changed
-------
* valid date formats for all date fields restricted 
* Schema and codelist titles and descriptions updated
* Documentation images and text updated 
* Documentation guidance restructured into modelling requirements and system requirements

> * Ownership-or-control statements replaced with Relationship statements

Reviewer: ```suggestion
* 'Ownership-or-control' statements renamed 'Relationship' statements
```

> * Statement fields:

Reviewer: ```suggestion
* Statement fields:

```
I think you need to include a blank line under this to make the sublist appear correctly in the built page.

  * ``statementDate`` now required
  * ``statementType`` replaced with ``recordType``
  * ``statementID`` renamed ``statementId``

> * Entity fields: 

Reviewer: ```suggestion
* Entity fields: 

```

  * ``entityType`` and ``entitySubtype`` combined into ``entityType``
  * ``securitiesListing`` removed from required fields in ``hasPublicListing``

>   * ``nomination`` and ``trust`` added to ``entitySubtype`` codelist

Reviewer: ```suggestion
  * 'nomination' and 'trust' added ``entitySubtype`` codelist
```

>   * prefixes removed from ``entitySubtype`` codelist (e.g. ``stateBody-stateAgency`` is now ``stateAgency``) 

Reviewer: ```suggestion
  * prefixes removed from ``entitySubtype`` codelist (e.g. 'stateBody-stateAgency' is now 'stateAgency') 
```

> * Relationship fields:

Reviewer: ```suggestion
* Relationship fields:

```

  * ``Interest/share/maximum`` is now inclusive by default. ``exclusiveMinimum`` and ``exclusiveMaximum`` are now numbers

>   * ``nominee`` and ``nominator`` added to interest type codelist 

Reviewer: ```suggestion
  * 'nominee' and 'nominator' added to ``Interest.type`` codelist 
```

>   * ``interestedParty.describedByEntityStatement``, ``interestedParty.describedByPersonStatement``, ``interestedParty.unspecified`` combined into ``interestedParty`` which is either a string referring to a recordId or an Unspecified Record. 

Reviewer: ```suggestion
  * ``interestedParty`` and ``subject`` fields now hold a ``recordId`` value (rather than a ``statementId`` value). They may instead take an Unspecified Record object, to represent missing information.
```

>   * ``componentStatementIds`` replaced with ``componentRecordIds``

Reviewer: ```suggestion
  * ``componentStatementIds`` renamed ``componentRecordIds`` and now holds ``recordId`` values
```

* Person fields:
  * ``fullName`` is now required 

>   * Renamed nameType code ``individual`` to ``legal`` 

Reviewer: ```suggestion
  * Renamed ``name.type`` code 'individual' to 'legal'
```

* ``address.country`` is now a Country object not a country code. 

Removed
-------
 * ``placeOfResidence``
 * ``agent``
 * ``replacesStatements``
 * Functional requirements page
 * Sources and annotations page

 
Alterations to schema structure and logic
-----------------------------------------

> * Updated the JSON Schema version from draft-04 to 2020-12

Reviewer: ```suggestion
* Updated the JSON Schema version from draft-04 to 2020-12
* Schema files renamed and contents refactored
```

* Included as much validation as possible within the schema 

> * id to $id

Reviewer: I think the id to $id and the def to $def change *may* be a byproduct of the move to JSON schema 2020-12. In which case we can remove these two lines.

* definitions to $defs

> * enum`s with one value are now `const`

Reviewer: ```suggestion
* enum types with one value replaced with const type
```

* Documentation images and text updated 
* Documentation guidance restructured into modelling requirements and system requirements
* 'Ownership-or-control' statements renamed 'Relationship' statements
* Statement fields:

  * ``statementDate`` now required
  * ``statementType`` replaced with ``recordType``
  * ``statementID`` renamed ``statementId``
* Entity fields: 

  * ``entityType`` and ``entitySubtype`` combined into ``entityType``
  * ``securitiesListing`` removed from required fields in ``hasPublicListing``
  * 'nomination' and 'trust' added ``entitySubtype`` codelist
  * prefixes removed from ``entitySubtype`` codelist (e.g. 'stateBody-stateAgency' is now 'stateAgency') 
* Relationship fields:

  * ``Interest/share/maximum`` is now inclusive by default. ``exclusiveMinimum`` and ``exclusiveMaximum`` are now numbers
  * 'nominee' and 'nominator' added to ``Interest.type`` codelist 
  * ``interestedParty`` and ``subject`` fields now hold a ``recordId`` value (rather than a ``statementId`` value). They may instead take an Unspecified Record object, to represent missing information.
  * ``componentStatementIds`` renamed ``componentRecordIds`` and now holds ``recordId`` values

> * Person fields:

Reviewer: ```suggestion
* Person fields:

```

The sublist isn't formatted properly. I think the extra line should sort that.
